### PR TITLE
feature/serving_history

### PR DESCRIPTION
**Pull request by at-thaile2**

### CODE REVIEW

File: Core/src/main/java/at/base/BasePage.java [hunk 1]
```diff
@@ -207,6 +206,7 @@ public BasePage clickBackButton() {
 
     /**
      * Navigate to page
+     * @return
```
Comment (at-anhquach) [inline Core/src/main/java/at/base/BasePage.java:209]: not change this file

File: Core/src/main/java/at/core/AppiumController.java [hunk 2]
```diff
@@ -102,15 +102,15 @@ void stop() {
 
     private synchronized void startDefaultServer() throws MalformedURLException {
         XmlTest xmlTest = new XmlTest();
-        xmlTest.setParameters(defaultIosParameters());
```
Comment (at-anhquach) [inline Core/src/main/java/at/core/AppiumController.java:105]: not change this file

File: Training/src/test/java/page/exam/HomePage.java [hunk 3]
```diff
@@ -51,6 +55,9 @@ public String getTextTvMenuToggleModes() {
         return tvMenuToggleModes.getText();
     }
 
+    public void clickStatButton(){
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/HomePage.java:58]: not change this file

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    private MobileElement screenTitle;
+
+    @iOSFindBy(id = "Daily Dozen")
+    @AndroidFindBy(xpath = "//*[@text='Daily Servings History']")
+    private MobileElement backButton;
+
+    @iOSFindBy(xpath = "//*[@class='UIAView']")
+    private MobileElement timeView;
+
+    @AndroidFindBy(xpath = "//*[@id='time_scale_spinner']")
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:29]: why not use id

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+public class ServingHistoryPage extends BasePage {
+    @iOSFindBy(id = "Servings History")
+    @AndroidFindBy(xpath = "//*[@contentDescription='Navigate up']")
+    private MobileElement screenTitle;
+
+    @iOSFindBy(id = "Daily Dozen")
+    @AndroidFindBy(xpath = "//*[@text='Daily Servings History']")
+    private MobileElement backButton;
+
+    @iOSFindBy(xpath = "//*[@class='UIAView']")
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:26]: use classname = UIAView

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+/**
+ * for screen
+ *
+ * @author at-thai.le
+ */
+
+public class ServingHistoryPage extends BasePage {
+    @iOSFindBy(id = "Servings History")
+    @AndroidFindBy(xpath = "//*[@contentDescription='Navigate up']")
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:19]: is not working

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    @Override
+    public boolean isPageDisplayed() {
+        return isForElementPresent(screenTitle);
+    }
+
+    @Override
+    public BasePage open() {
+        if (!isPageDisplayed()) {
+            HomePage hp = new PageFactory<>(HomePage.class).create();
+            getDriver().launchApp();
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:81]: not code this funtion

File: Training/src/test/resources/features/at_thaile/ServingHistory.feature [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    When I click Day button
+    Then Day button is active
+
+  Scenario: Check click Setting button
+    When I click Setting button
+    Then Display Setting screen
+
+  Scenario: Check click Info button
+    When I click Info button
+    Then Display Info screen
```
Comment (at-anhquach) [inline Training/src/test/resources/features/at_thaile/ServingHistory.feature:40]: enter line

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    private MobileElement monthButton;
+
+    @iOSFindBy(id = "Year")
+    @AndroidFindBy(xpath = "//*[@text='Year']")
+    private MobileElement yearButton;
+
+    @iOSFindBy(id = "ic left")
+    private MobileElement leftButton;
+
+    @iOSFindBy(id = "ic left double")
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:47]: where Android?

File: Training/src/test/java/page/exam/HomePage.java (hunk 3 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/HomePage.java:58]: Got it

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java (hunk 9 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:47]: Android isn't the same as iOS

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java (hunk 7 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:81]: Just coding for testing, sorry

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java (hunk 5 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:26]: updated, thank you

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java (hunk 4 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:29]: updated, thank you

File: Core/src/main/java/at/base/BasePage.java (hunk 1 above)
Comment (at-thaile2) [reply Core/src/main/java/at/base/BasePage.java:209]: Got it

File: Core/src/main/java/at/core/AppiumController.java (hunk 2 above)
Comment (at-thaile2) [reply Core/src/main/java/at/core/AppiumController.java:105]: Just changed for testing, sorry

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java (hunk 6 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:19]: Fixed

File: Training/src/test/java/page/exam/HomePage.java [hunk 10]
```diff
@@ -21,6 +22,9 @@
     @AndroidFindBy(className = "android.widget.ListView")
     private MobileElement listMenu;
 
+    @iOSFindBy(id = "ic stat")
+    private MobileElement statButton;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/HomePage.java:26]: don't change this file

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+/**
+ * for screen
+ *
+ * @author at-thai.le
+ */
+
+public class ServingHistoryPage extends BasePage {
+    @iOSFindBy(id = "Servings History")
+    @AndroidFindBy(xpath = "//*[@text='Daily Servings History']")
+    private MobileElement screenTitle;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:21]: tvTitle

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+    @iOSFindBy(id = "21 Tweaks")
+    @AndroidFindBy(className = "android.widget.ImageButton")
+    private MobileElement backButton;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:25]: btnBack

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+    @iOSFindBy(id = "Time Scale")
+    @AndroidFindBy(xpath = "//*[@text='Time Scale']")
+    private MobileElement scaleText;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:29]: tvScale

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+
+    @iOSFindBy(id = "Day")
+    @AndroidFindBy(xpath = "//*[@text='Day']")
+    private MobileElement dayButton;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:33]: btnDay

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+    @iOSFindBy(id = "Month")
+    @AndroidFindBy(xpath = "//*[@text='Month']")
+    private MobileElement monthButton;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:37]: btnMonth

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+
+    @iOSFindBy(id = "Year")
+    @AndroidFindBy(xpath = "//*[@text='Year']")
+    private MobileElement yearButton;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:41]: btnYear

File: Training/src/test/resources/features/exam/Home.feature [hunk 17]
```diff
@@ -11,6 +11,11 @@ Feature: Test on Home screen
     When I check content tweaks button
     Then The content contain "TWEAKS"
 
+  Scenario: Check click Daily Dozen button
+    When I click Daily Dozen button
+    Then Display Daily Dozen screen
```
Comment (at-anhquach) [inline Training/src/test/resources/features/exam/Home.feature:16]: don't change this file

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java (hunk 11 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:21]: Fixed

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java (hunk 12 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:25]: Fixed

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java (hunk 13 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:29]: Fixed

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java (hunk 14 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:33]: Fixed

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java (hunk 15 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:41]: Fixed

File: Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java (hunk 16 above)
Comment (at-thaile2) [reply Training/src/test/java/page/exam/at_thaile/ServingHistoryPage.java:45]: Fixed

**Pull request merged**